disco_aws_automation/disco_vpc.py

```diff
@@ -433,36 +463,52 @@ def create(self):
             if vpc_cidr is None:
                 raise VPCConfigError('Cannot create VPC %s. No subnets available' % self.environment_name)
 
-        # Create VPC
+        return vpc_cidr
+
+    def _create_vpc(self):
+        """
+        Create a new VPC and add the default and custom tags
+        :param vpc_cidr:
+        :return:
+        """
+
+        # Get the vpc CIDR
+        vpc_cidr = self._get_vpc_cidr()
+
+        # Create the new VPC
         self.vpc = throttled_call(self.boto3_ec2.create_vpc, CidrBlock=str(vpc_cidr))['Vpc']
         waiter = self.boto3_ec2.get_waiter('vpc_available')
         waiter.wait(VpcIds=[self.vpc['VpcId']])
+
+        # Add tags to VPC
+        self._add_vpc_tags()
+
+        logger.debug("vpc: %s", self.vpc)
+
+    def _add_vpc_tags(self):
+        """
+        Add tags to VPC. This function will add the default tags and the tags specified on the create
+        vpc command
+        """
+        # get the resource representing the new VPC
         ec2 = boto3.resource('ec2')
         vpc = ec2.Vpc(self.vpc['VpcId'])
-        tags = vpc.create_tags(Tags=[{'Key': 'Name', 'Value': self.environment_name},
-                                     {'Key': 'type', 'Value': self.environment_type}])
-        logger.debug("vpc: %s", self.vpc)
-        logger.debug("vpc tags: %s", tags)
 
-        # Enable DNS
-        throttled_call(self.boto3_ec2.modify_vpc_attribute,
-                       VpcId=self.vpc['VpcId'], EnableDnsSupport={'Value': True})
-        throttled_call(self.boto3_ec2.modify_vpc_attribute,
-                       VpcId=self.vpc['VpcId'], EnableDnsHostnames={'Value': True})
+        tag_list = [{'Key': 'Name', 'Value': self.environment_name},
+                    {'Key': 'type', 'Value': self.environment_type},
+                    {'Key': 'create_date', 'Value': datetime.utcnow().isoformat()}]
 
-        self._networks = self._create_new_meta_networks()
-        if not self.skip_enis_pre_allocate:
-            self._reserve_hostclass_ip_addresses()
+        if self._vpc_tags:
+            # Convert the list of tags key:value strings (example ["productline:myproduct", ...])
+            # into a list of dictionary as expected by the vpc.create_tags function
+            # (example: [{'Key': 'productline', 'Value': 'myproduct'}, ...]
+            extra_vpc_tags = dict2tag(self._vpc_tags)
```

Comment:
Would it make more sense to do this parsing earlier, possibly in the bin file? Then if there was a problem parsing user input, we wouldn't have to worry about cleaning up a half-created VPC.